### DISCUSSION TITLE
How to use the KnowWE wiki pages as a standalone application?

### DISCUSSION THREAD

**Discussion by bobdavis62** [denkbares/d3web-KnowWE#16] [Q&A]
Wow! What a fantastic tool you have created. Thank you so much for making it available open-source.

I have spent the last couple days working with KnowWE to create an expert system to trouble-shoot faults with VMEbus computer systems (patterned after the car diagnostics demo). I really like the wiki-based interface and think it would be more than adequate for my application. I have read in the KnowWE developer FAQ that it is possible to simply "publish" this wiki.

I think I know what is required to "publish" the wiki. I think that I just have to copy the KnowWE folder (and all subfolders) to the Tomcat/webapps folder and then when I run Tomcat again it will recognize a new web app has been installed. Is this more or less correct?

Once I "publish" the wiki pages I want to turn-off or disable some of the KnowWE features like page editing and summary information like the rules and other statistics that is helpful to the designer of the wiki, but not needed by the user of the wiki. So, basically, I would like to be left with just the interview feature and the decision tree.

Is it possible to do this? I'd like to try to avoid actually writing a Java program and calling the KnowWE libraries from the Java code if possible.

Again, thanks for a great product and hope to hear from someone soon!

Bob

**Comment by AlbrechtStriffler** [comment]
Hi Bob,
sorry for the late answer, the notification got burried in my inbox...

With publishing, we actually refer to downloading the knowledge base file (which is just a zip file containing xmls) in the %%KnowledgeBase markup and loading that file in some other application, that usually has a GUI build with a certain purpose in mind.

What you suggested should also be possible though and there are some variants...
In your implementation, did you use different pages and packages to create the knowledge base? If yes, you can always just make a clean new page with just the %%QuickInterview and maybe the %%ShowSolutions-Markup... it will show the right content, as long as it uses the same package as the %%KnowledgeBase markup. As an example, see this page: https://www.d3web.de/Wiki.jsp?page=Demo%20-%20Master%20-%20Interview 
As long as you don't have the rules and other markups linked on the main page or left menu and as long as the users don't have write permissions, this should work fine.

Using some jspwiki-markups and permissions, you could also actively restrict the access of normal users to the sites containing the knowledge and rules you don't want to show, check out https://jspwiki-wiki.apache.org/Wiki.jsp?page=Wiki.Admin.Security for more information on that. As an example, you could add the text `[{ALLOW view Bob}]` at the top of the page, to restrict its visibility to just yourself (in case you are registered as Bob in your instance of KnowWE).

Hope this helps
Albrecht
